### PR TITLE
Implement upgrade test for Training operator

### PR DESCRIPTION
**Pull request by sutaakar**
## Description

Test verifies that Training operator is able to monitor PyTorchJob created by previous version.
Moved base tests into core package to separate it from upgrade tests.

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by astefanutti — COMMENTED** [review]
Just to understand it, that's the training operator that's being upgraded and that it's working with existing resources?

**Comment by sutaakar** [comment]
> Just to understand it, that's the training operator that's being upgraded and that it's working with existing resources?

Yes

File: tests/kfto/core/environment.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ const (
 )
 
 func GetFmsHfTuningImage() string {
-	return lookupEnvOrDefault(fmsHfTuningImageEnvVar, "quay.io/modh/fms-hf-tuning:d0bd35b0297c28b87ee6caa32d5966d77587591f")
+	return lookupEnvOrDefault(fmsHfTuningImageEnvVar, "quay.io/modh/fms-hf-tuning:bd8bf628cd739c7a201a976bc3c1096785353f1a")
```
Comment (jiripetrlik) [inline tests/kfto/core/environment.go:31]: Isn't there a better way how to parameterize this?

**Review by jiripetrlik — APPROVED** [review]

**Pull request merged**